physcraper/__init__.py

```diff
@@ -1981,6 +2017,39 @@ def read_local_blast_query(self, fn_path):
                                                query_dict[key]["evalue"]))
                 fn.close()
 
+    def get_taxid_from_acc(self, gb_acc):
+        """
+        Use the blastdb to get the taxon id from a queried gb acc. 
+
+        Sometimes there are more than a singel id, as database has merged redundant seqs.
+
+        :param gb_acc: Genbank accession number
+        :return: list of taxon ids associated with the GB id - there are multiple because of the merging of redundant data
+        """
+        if not os.path.exists("{}/tmp".format(self.workdir)):
+            os.mkdir("{}/tmp".format(self.workdir))
+        fn = "{}/tmp/tmp_search.csv".format(self.workdir)
+        fn_open = open(fn, "w+")                                        
+        fn_open.write("{}\n".format(gb_acc))
+        fn_open.close()
+        cmd1 = "blastdbcmd -db {}/nt  -entry_batch {} -outfmt %T -out {}/tmp/tax_id_{}.csv".format(self.config.blastdb, fn, self.workdir, gb_acc)
+        if not self.config.blastdb == "./tests/data/precooked/testing_localdb":
+            os.system(cmd1)
+            f = open("{}/tmp/tax_id_{}.csv".format(self.workdir, gb_acc))
+            tax_id_l = []
+            for line in iter(f):
+                line = line.rstrip().lstrip()
+                tax_id_l.append(int(line))
+            f.close() 
+        else: 
+            f = open("./tests/data/precooked/testing_localdb/tax_id_{}.csv".format(gb_acc))
```

Comment:
I'm not sure what exactly is happening here, but I don't think we want to hardcode in something from testing as the blast database.

Reply:
I implemented this so that we can still run tests. This function relies on the blast db and as we do not have them, we need to get the stored files from the fprecooked folder.

Reply:
Ideally in testing we are running the same commands as when it isn't for testing. It looks like it is creating a csv for each accession number with taxon ids that go with that accession number, and then looks for them in workdir/tmp. If that is correct, one way to make it work the same for the tests is to copy the pre-cooked 'tax_id_gb_acc.csv' into tmp/workdir as part of the test, rather than hardcoding it into the code. Do you think that will work?